demo/api/src/importer/data-streams/data-stream.ts

```diff
@@ -1,9 +1,11 @@
-export interface DataStreamAndMetadata {
-    dataStream: NodeJS.ReadableStream;
-    params?: Record<string, unknown>;
-    additionalData?: Record<string, unknown>;
-}
+import { Readable } from "stream";
+
+import { PipeMetadata } from "../pipes/importer-pipe.type";
 
+export type StreamChunkAndMetadata = {
+    chunk: Buffer | string;
+    metadata: PipeMetadata<Record<string, unknown>>;
```

Comment:
We probably should be able to specify the shape of the metadata everywhere by using generics. We can discuss this tomorrow.